### PR TITLE
[QA] Add PHP-CS-Fixer 

### PR DESCRIPTION
**Pull request by veewee**
Automatically check some CS rules as described in #30 

FYI : the config and the fixes are in 2 separate commits. I can change those independently if you wish to change a setting.

### CODE REVIEW

**Comment by veewee** [comment]
The latest psalm version gives this issue:

```
ERROR: RedundantCondition - src/Psl/Str/convert_encoding.php:27:5 - Type string for $result is never false (see https://psalm.dev/122)
    Psl\invariant(false !== $result, 'Unable to convert $string from $from_encoding to $to_encoding.');


ERROR: RedundantCondition - src/Psl/Str/convert_encoding.php:27:19 - string can never contain false (see https://psalm.dev/122)
    Psl\invariant(false !== $result, 'Unable to convert $string from $from_encoding to $to_encoding.');


```

Haven't touched the file. Is it safe to remove the invariant check? Or shall I add an ignore for that rule?
Looks like it always returns a string: https://www.php.net/manual/en/function.mb-convert-encoding.php

**Comment by azjezz** [comment]
okay; you can go ahead and remove that check.

File: src/Psl/Exception/InvariantViolationException.php [hunk 1]
```diff
@@ -4,6 +4,6 @@
 
 namespace Psl\Exception;
 
-final class InvariantViolationException extends RuntimeException
+class InvariantViolationException extends RuntimeException
```
Comment (azjezz) [inline src/Psl/Exception/InvariantViolationException.php]: why was final removed from this class?

Comment (veewee) [reply src/Psl/Exception/InvariantViolationException.php]: Manual mistake, re-added it.

**Review by azjezz — APPROVED** [review]

**Pull request merged**